jiant/proj/main/modeling/taskmodels.py

```diff
@@ -276,14 +282,94 @@ def get_output_from_encoder(encoder, input_ids, segment_ids, input_mask) -> Enco
         EncoderOutput containing pooled and unpooled model outputs as well as any other outputs.
 
     """
+    model_arch = ModelArchitectures.from_encoder(encoder)
+    if model_arch in [
+        ModelArchitectures.BERT,
+        ModelArchitectures.ROBERTA,
+        ModelArchitectures.ALBERT,
+        ModelArchitectures.XLM_ROBERTA,
+    ]:
+        pooled, unpooled, other = get_output_from_standard_transformer_models(
+            encoder=encoder, input_ids=input_ids, segment_ids=segment_ids, input_mask=input_mask,
+        )
+    elif model_arch in [
+        ModelArchitectures.XLM,
+    ]:
+        pooled, unpooled, other = get_output_from_xlm_with_lang_handing(
+            encoder=encoder, input_ids=input_ids, input_mask=input_mask, batch=batch,
+        )
+    elif model_arch in [
+        ModelArchitectures.BART,
+        ModelArchitectures.MBART,
+    ]:
+        pooled, unpooled, other = get_output_from_bart_models(
+            encoder=encoder, input_ids=input_ids, input_mask=input_mask,
+        )
+    else:
+        raise KeyError(model_arch)
```

Comment:
might be cleaner to move the if/else on `model_arch` logic down a layer since `get_output_from*` already has an `encoder` argument. Then all, `get_output_from*` can be collapsed into one function.